cmd/cloudFoundryDeploy.go

```diff
@@ -0,0 +1,855 @@
+		err := fileUtils.FileWrite(smokeTestScript, []byte(defaultSmokeTestScript), 0755)
+		if err != nil {
+			return []string{}, err
+		}
+		log.Entry().Debugf("smoke test script '%s' has been written.", smokeTestScript)
+	}
+
+	if len(smokeTestScript) > 0 {
+		err := fileUtils.Chmod(smokeTestScript, 0755)
+		if err != nil {
+			return []string{}, err
+		}
+		pwd, err := fileUtils.Getwd()
+		if err != nil {
+			return []string{}, err
+		}
+
+		return []string{"--smoke-test", fmt.Sprintf("%s/%s", pwd, smokeTestScript)}, nil
+	}
+	return []string{}, nil
+}
+
+func prepareBlueGreenCfNativeDeploy(config *cloudFoundryDeployOptions) (string, []string, []string, error) {
+
+	smokeTest, err := handleSmokeTestScript(config.SmokeTestScript)
+	if err != nil {
+		return "", []string{}, []string{}, err
+	}
+
+	var deployOptions = []string{}
+
+	if !config.KeepOldInstance {
+		deployOptions = append(deployOptions, "--delete-old-apps")
+	}
+
+	if len(config.Manifest) > 0 {
+		manifestFileExists, err := fileUtils.FileExists(config.Manifest)
+		if err != nil {
+			return "", []string{}, []string{}, err
+		}
+
+		if !manifestFileExists {
+
+			log.Entry().Infof("Manifest file '%s' does not exist", config.Manifest)
+
+		} else {
+
+			manifestVariables, err := toStringInterfaceMap(toParameterMap(config.ManifestVariables))
+			if err != nil {
+				return "", []string{}, []string{}, err
+			}
+
+			manifestVariablesFiles, err := removeDefaultManifestVariableFileIfItIsTheOnlyFileAndThatFileDoesNotExist(config.ManifestVariablesFiles)
+			if err != nil {
+				return "", []string{}, []string{}, err
+			}
+
+			modified, err := _substitute(config.Manifest, manifestVariables, manifestVariablesFiles)
+			if err != nil {
+				return "", []string{}, []string{}, errors.Wrap(err, "Cannot prepare manifest file")
+			}
+
+			if modified {
+				log.Entry().Infof("Manifest file '%s' has been updated (variable substitution)", config.Manifest)
+			} else {
+				log.Entry().Infof("Manifest file '%s' has not been updated (variable substitution)", config.Manifest)
+			}
+
+			err = handleLegacyCfManifest(config.Manifest)
+			if err != nil {
+				return "", []string{}, []string{}, err
+			}
+		}
+	} else {
+		log.Entry().Info("No manifest file configured")
+	}
+	return "blue-green-deploy", deployOptions, smokeTest, nil
+}
+
+func removeDefaultManifestVariableFileIfItIsTheOnlyFileAndThatFileDoesNotExist(manifestVariablesFiles []string) ([]string, error) {
+
+	const defaultManifestVariableFileName = "manifest-variables.yml"
+	if len(manifestVariablesFiles) == 1 && manifestVariablesFiles[0] == defaultManifestVariableFileName {
+		// we have only the default file. Most likely this is not configured, but we simply have the default.
+		// In case this file does not exist we ignore that file.
+		exists, err := fileUtils.FileExists(defaultManifestVariableFileName)
+		if err != nil {
+			return []string{}, err
+		}
```

Comment:
```suggestion
		exists, _ := fileUtils.FileExists(defaultManifestVariableFileName)
```

... and then don't return an error from this function at all.

Reply:
It's not cool to drop exceptions silently. In the unlikely but possible case that is not possible to check if a file exists we have _severe_ trouble and we should no continue with a "business as usual" attitude. I introduced exception wrapping.

Reply:
My suggestion was specific to the context of this function and FileExists() in particular. Please don't judge it without that context. In general I am all for proper error handling. For the code-flow, handling an error from FileExists() often just gives a more specific error message. It does not mean that business continues as usual. In any case, wrapping the error to achieve the effect better location of the error from the log, is just as good a solution.